src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,43 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Locks the current handle and reads a line of input, returning a `String` containing
+/// the input.
+/// 
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// [`Stdin::lock`]: struct.Stdin.html#method.lock
+/// [`Stdout::lock`]: struct.Stdout.html#method.lock
+/// 
+/// ### Note: Windows Portability Consideration
+/// When operating in a console, the Windows implementation of this stream does not support
+/// non-UTF-8 byte sequences. Attempting to read and write bytes that are not valid UTF-8 will 
+/// return an error.
+///
+/// # Examples
+///
+/// ```no_run
+/// use std::io::input;
+///
+/// fn main() {
+///     let user_input = input("Please enter some text: ");
+///         
+///     println!("You typed: {}", get);
+///         
+/// }
+/// ```
+pub fn input(message:&str) -> String {
+    print!("{}", message);
```

Comment:
I don't think we need a macro `print` here, we could just use the underlying layer. (probably compiler speed improvement, me guessing)

Reply:
I removed the ```print!()``` macro